src/dependency-manager/src/service-config/base.ts

```diff
@@ -65,6 +72,7 @@ export abstract class ServiceConfig {
   abstract getName(): string;
   abstract getLanguage(): string;
   abstract getImage(): string;
+  abstract getDockerOptions(): ServiceDockerSpec;
```

Comment:
This feels redundant since we already have command/entrypoint at the top level. See two lines below.

Think I'd prefer:
```
  "command": "node index.js"
  "debug": {
    "dockerfile": ".docker/Dockerfile",
    "context": "./src/",
    "command": "echo \"DEBUG\""
    ...
  }
```

Dockerfile/context are weird since they only apply to building the image and won't apply to k8s/ecs. Also weird because we also have the platforms block for defining platform specific config.

Reply:
I thought about doing that to flatten the relationship, but then thought it might be weird with volumes or other non-docker references in the debug block. For example, we might want to define volumes in the debug block like we discussed earlier:

```json
{
  "debug": {
    "docker": {
      "//": "..."
    },
    "volumes": {
      "watch-directory": {
        "description": "Directory containing source code to be mounted locally so files can be watched and updated",
        "mountPath": "./src"
      }
    }
  }
}
```

Do you prefer the above with or without the docker sub-block?